app/models/concerns/media_facebook_profile.rb

```diff
@@ -11,93 +11,116 @@ module MediaFacebookProfile
+    end
 
-  def get_data_from_facebook
     data = {}
-    id = self.get_facebook_id_from_url
-    client = self.facebook_client
-    self.data[:raw][:api] = {}
     # Try to parse as a user profile first
     begin
-      object = client.get_object(id, { fields: self.facebook_user_fields }, { method: 'post' })
-      self.data[:raw][:api] = object
+      data = self.parse_facebook_user
       data['subtype'] = 'user'
     # If it fails, try to parse as a page
     rescue
-      object = client.get_object(id, { fields: self.facebook_page_fields }, { method: 'post' })
-      self.data[:raw][:api] = object
+      begin
+        data = self.parse_facebook_page
+      rescue
+        data = self.parse_facebook_legacy_page
+      end
       data['subtype'] = 'page'
     end
+    data['id'] = self.get_facebook_id_from_url
+
+    error = self.get_facebook_profile_error
+    data['error'] = error if error
+    
+    data['likes'] = self.get_facebook_likes
+    
     data['published_at'] = ''
     data
   end
 
-  def data_from_facebook_profile
-    handle_exceptions(self, Koala::Facebook::ClientError, :fb_error_message, :fb_error_code) do
-      begin
-        self.data.merge! self.get_data_from_facebook
-      rescue Koala::Facebook::ClientError => e
-        Rails.logger.info "[Facebook Profile] Parsing `#{url}`: Error Code: #{e.fb_error_code} Subcode #{e.fb_error_subcode} Message: #{e.fb_error_message}"
-        Airbrake.notify(e) if Airbrake.configuration.api_key && e.fb_error_code == 190
-        raise e
-      end
+  def get_facebook_profile_error
+    page = self.get_facebook_profile_page
+    title = page.css('meta[property="og:title"]')
+    if title.present? && title.attr('content') && title.attr('content').value == 'Log In or Sign Up to View'
+      { message: 'Login required to see this profile' }
+    end
+  end
+
+  def get_facebook_profile_html
+    if @html.nil?
+      @html = self.get_html(self.html_options).to_s
+    end
+    @html
+  end
+
+  def get_facebook_profile_page
+    if @page.nil?
+      @page = self.get_html(self.html_options)
     end
-    self.get_facebook_likes
+    @page
+  end
+
+  def parse_facebook_user
+    page = self.get_facebook_profile_page
+
+    data = {}
+    data['name'] = page.css('#fb-timeline-cover-name').first.text
+    bio = page.css('#pagelet_bio span').last
+    data['description'] = bio ? bio.text : ''
+    data['picture'] = page.css('.img.profilePic.img').first.attr('src')
+    data
+  end
+
+  def parse_facebook_page
+    html = self.get_facebook_profile_html
+    page = self.get_facebook_profile_page
+    match = html.match(/"name":"([^"]+)","pageID":"([^"]+)","username":([^,]+),"usernameEditDialogProfilePictureURI":"([^"]+)"/)
+    json = JSON.parse('{' + match[0] + '}')
+    
+    data = {}
+    data['name'] = json['name']
+    data['username'] = json['username']
+    data['description'] = page.css('meta[name=description]').first.attr('content').gsub(/.*talking about this. /, '')
```

Comment:
are we ensuring English language here?

Reply:
yes, I set the header `Accept-Language: en` in the crawler request

Reply:
looks like we are below, yes